compiler/rustc_infer/src/infer/outlives/verify.rs

```diff
@@ -286,16 +286,6 @@ impl<'cx, 'tcx> VerifyBoundCx<'cx, 'tcx> {
     /// the trait. In fact, it works by searching for just such a
     /// where-clause.
     ///
-    /// It will not, however, work for higher-ranked bounds like:
-    ///
-    /// ```compile_fail,E0311
-    /// trait Foo<'a, 'b>
-    /// where for<'x> <Self as Foo<'x, 'b>>::Bar: 'x
-    /// {
-    ///     type Bar;
-    /// }
-    /// ```
-    ///
```

Comment:
This can be compiled without any problem.

Reply:
The sentence below is still wrong and I'd like to get confirmation from people knowing this code better that this is indeed good and correct and whether the docs should be adjusted further

Reply:
Can we switch this to an ignore test and add a comment referencing that this does compile today? I think that would be better than deleting this test and allow us to move forward with this PR.